src/minion-ffi/tests/runner.c

```diff
@@ -0,0 +1,162 @@
+#include "minion-ffi.h"
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#define __call_where(f, l, s, ...) f(__FILE__ ":" #l ": `" #s "`", __VA_ARGS__)
+#define _call_where(f, l, ...) __call_where(f, l, f(__VA_ARGS__), __VA_ARGS__)
+#define call_where(f, ...) _call_where(f, __LINE__, __VA_ARGS__)
+
+static void die(char const* message) {
```

Comment:
`die` should be `printf`-like function. All usages of `fprintf`+`abort` should be switched to die.